AxonDeepSeg/data_management/dataset_building.py

```diff
@@ -247,3 +248,50 @@ def find_minority_type(SEM_patches_folder, TEM_patches_folder):
         len_majority = SEM_len
 
     return minority_patches_folder, len_minority, majority_patches_folder, len_majority
+
+def split_data(raw_dir, out_dir, seed=2019, split = [0.8, 0.2], override=False):
```

Comment:
I think one better way to split the dataset is during training itself using Keas API. For more details look at keras `model.fit `method , refer here https://keras.io/models/model/. Here using` validation split =0.2 `, we can split the data into training 80 % of the data and validation 20 % of the data. For additional details on how the shuffling and spilt is performed, https://stackoverflow.com/questions/51006505/how-training-and-test-data-is-split-keras-on-tensorflow#comment89010893_51006505 is good resource. Personally, instead of writing our own method we could use splitting API of keras during training.  @mathieuboudreau, let me know what  you think about it ?

Reply:
Also note, I don't know whether this will create directories for train and validation separately. And, also, if we think of implementing it, some small changes into how  training the models had to be  carried out.

Reply:
In that case, maybe we should move ahead with my implementation for the time being, as your other branch is being held up by the OSF i/o (needs this feature to work smoothly in notebooks). We could then open an issue and look into your proposed method later on. What do you think @jcohenadad